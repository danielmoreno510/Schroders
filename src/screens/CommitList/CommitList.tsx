import React from 'react';
import {View, FlatList} from 'react-native';

import Commit from '../../components/common/Commit';
import styles from './styles';

const RepositoryList: React.FC<ICommitListProps> = ({
  commitList,
  setCurrentCommit,
  navigation,
}) => {
  const showCommit = (commit: ICommit) => {
    navigation.navigate('Details');
    setCurrentCommit(commit);
  };

  return (
    <View style={styles.container}>
      <FlatList
        data={commitList}
        renderItem={({item}) => (
          <Commit commit={item} showCommit={showCommit} />
        )}
        keyExtractor={item => item.node_id}
        showsVerticalScrollIndicator={false}
      />
    </View>
  );
};

export default RepositoryList;
