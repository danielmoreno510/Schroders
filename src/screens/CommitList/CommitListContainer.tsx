import CommitList from './CommitList';
import {StoreService} from '../../redux/StoreService';
import {putCurrentCommit} from '../../redux/actions';

export const mapStateToProps = (state: ICommitListStateProps) => ({
  commitList: state.commits.commitList,
});

export const mapDispatchToProps = (dispatch: (T: ICommitReducer) => void) => ({
  setCurrentCommit: (commit: ICommit) => {
    dispatch(putCurrentCommit(commit));
  },
});

const CommitListContainer = StoreService.connect(
  mapStateToProps,
  mapDispatchToProps,
)(CommitList);

export default CommitListContainer;
