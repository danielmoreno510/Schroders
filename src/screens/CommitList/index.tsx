import CommitListContainer from './CommitListContainer';

export default CommitListContainer;
