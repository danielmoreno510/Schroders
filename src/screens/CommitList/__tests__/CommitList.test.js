import React from 'react';
import TestRenderer from 'react-test-renderer';

import CommitList from '../CommitList';

jest.spyOn(React, 'useEffect').mockImplementation(f => f());

describe('CommitList component', () => {
  describe('GIVEN a set of valid props', () => {
    const testProps = {
      isFetchingCommitList: true,
      commitList: [
        {
          id: 1,
          node_id: '1',
          title: 'Test 1',
          user: {login: 'Test', id: '1', avatar_url: ''},
          comments: 3,
          created_at: '2022-09-21',
          body: 'Body 1',
        },
      ],
      getCommits: jest.fn(),
      setCurrentCommit: jest.fn(),
      navigation: {navigate: jest.fn()},
    };

    describe('WHEN the component is rendered', () => {
      const testRenderer = TestRenderer.create(<CommitList {...testProps} />);

      it('THEN should display a regular CommitList', () => {
        expect(testRenderer.toJSON()).toMatchSnapshot();
      });
    });

    describe('WHEN showCommit functions are called', () => {
      const customProps = {
        ...testProps,
        isFetchingCommitList: false,
        commitList: [
          {
            id: 1,
            node_id: '1',
            title: 'Test 1',
            user: {login: 'Test', id: '1', avatar_url: ''},
            comments: 3,
            created_at: '2022-09-21',
            body: 'Body 1',
          },
        ],
      };
      const testRenderer = TestRenderer.create(<CommitList {...customProps} />);
      const testInstance = testRenderer.root;
      testInstance.findByProps({testID: 'showCommit'}).props.onPress();

      it('THEN setCurrentCommit function should be called', () => {
        expect(testProps.setCurrentCommit).toBeCalledTimes(1);
      });
    });
  });
});
