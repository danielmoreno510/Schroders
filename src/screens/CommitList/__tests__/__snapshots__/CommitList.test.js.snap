// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommitList component GIVEN a set of valid props WHEN the component is rendered THEN should display a regular CommitList 1`] = `
<View
  style={
    {
      "backgroundColor": "white",
      "flex": 1,
      "padding": 37.5,
    }
  }
>
  <RCTScrollView
    data={
      [
        {
          "body": "Body 1",
          "comments": 3,
          "created_at": "2022-09-21",
          "id": 1,
          "node_id": "1",
          "title": "Test 1",
          "user": {
            "avatar_url": "",
            "id": "1",
            "login": "Test",
          },
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsVerticalScrollIndicator={false}
    stickyHeaderIndices={[]}
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            {
              "alignContent": "center",
              "borderBottomColor": "#f1b30a",
              "borderBottomWidth": 1,
              "flex": 1,
              "justifyContent": "space-between",
              "padding": 0,
              "paddingHorizontal": 0,
            }
          }
        >
          <Text
            style={
              {
                "flexDirection": "column",
                "paddingTop": 8,
              }
            }
          >
            Author: 
            Test
          </Text>
          <Text
            style={
              {
                "flexDirection": "column",
                "paddingTop": 8,
              }
            }
          >
            Date: 
            2022-09-21
          </Text>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "backgroundColor": "#f1b30a",
                "borderRadius": 8,
                "margin": 12,
                "marginHorizontal": 0,
                "opacity": 1,
                "padding": 6,
                "width": 100,
              }
            }
            testID="showCommit"
          >
            <Text
              style={
                {
                  "color": "white",
                }
              }
            >
              Go To Details
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
